docs/modules/cloud/controller.md

```diff
@@ -0,0 +1,54 @@
+# Edge Controller
+
+
+## Edge Controller Overview
+ Controller(also known as edgecontroller) is the bridge between Kubernetes Api-Server and Edgecore
+
+   
+## Operations Performed By Edge Controller
+ 
+ The following are the functions performed by Edge controller :-
+  - Downstream Controller:Sync add/update/delete event to edgeCore from K8s Api-server
+  - Upstream Controller:sync watch and Update status of resource and events(node, pod and configmap) to K8s Api-server and also subscribe message from edgecore
+  - Controller Manager:Creates manager Interface which implements events for managing ConfigmapManager LocationCache and podManager
+
+
+### Downstream Controller:Sync add/update/delete event to edge
+ 
+- Downstream controller:Watches K8S-Api-server and send updates to edge via cloudHub
+- Sync (pod,configmap,secret) add/update/delete event to edge via cloudHub
+- Creates Respective manager (pod,configmap,secret) for handling events by calling manager interface
+- Locates configmap and secret should be send to which node
+
+![Downstream Controller](./../../images/edgecontroller/DownstreamController.png)
+### Upstream Controller:sync watch and Update status of resource and events
+
+- UpstreamController receives messages from edge and sync the updates to K8S-Api-server
+- Creates stop channel to dispatch and stop event to handle pods, configMaps ,node and secrets
+- Creates message channel to update Nodestatus, Podstatus, Secret and configmap related events
+- Gets Podcondition information like Ready, initialized, Podscheduled and Unschedulable details
+- **Below is the information for PodCondition**
+   - **Ready**:PodReady means the pod is able to service requests and should be added to the load balancing pools for all matching services
+   - **PodScheduled**:It represents status of the scheduling process for this pod
+   - **Unschedulable**:It means scheduler cant schedule the pod right now,may be due to insufficient resources in the cluster
+   - **Initialized**:It means that all Init containers in the pod have started sucessfully
+- Gives details about the Container status for the specific podsives 
+- Gives PodStatus information
+- **Below is the information for PodStatus**
+   - **PodPhase**:Current condition of the pod
+   - **Conditions**:Details indicating wht the pod is in this condition
+   - **HostIP**:Ip address of the host to which pod is assigned
+   - **PodIp**:Ip address allocated to the Pod
+   - **QosClass**:Assigned to the pod based on resource requirement
+   
+   ![Upstream Controller](./../../images/edgecontroller/UpstreamController.png)
+   
+### Controller Manager:Creates manager Interface and implements ConfigmapManager LocationCache and podManager
+
+- Manager define the Interface of a manager,ConfigManager,Podmanager,secretmanager implements it
```

Comment:
Please rephrase this statement.  
define -> defines  
nit: Spaces after commas.